core/dbt/task/test.py

```diff
@@ -126,6 +126,8 @@ def before_execute(self):
     def execute_data_test(self, data_test: TestNode, manifest: Manifest) -> TestResultData:
         context = generate_runtime_model_context(data_test, self.config, manifest)
 
+        hook_ctx = self.adapter.pre_model_hook(context)
```

Comment:
let's also test this codepath (hooks before/after data test execution) in addition to hooks before/after unit test execution. 

We don't already have a non-adapter `tests/functional/data_tests/` but let's create one! 

Here's some example setup for a singular data test that should run through this codepath: https://github.com/dbt-labs/dbt-core/blob/main/tests/functional/test_singular_tests.py

Beyond the setup, I believe the tests should look pretty similar to what we've got in test_ut_adapter_hooks.py so far

Reply:
added tests please let me know what you think! and thanks for all the suggestions @MichelleArk